packages/react/src/useActionForm.ts

```diff
@@ -101,6 +128,198 @@ const disambiguateDefaultValues = (data: any, initialData: any, action: any) =>
   return result;
 };
 
+const reshapeDataForGraphqlApi = async (client: AnyClient, defaultValues: any, data: any) => {
+  const referencedTypes = client[$modelRelationships];
+
+  if (!referencedTypes) {
+    throw new Error("No referenced types found");
+  }
+
+  function getUpdates(data: any): Record<string, number[]> {
+    const updateList: Record<string, number[]> = {};
+
+    function traverse(input: any, path: string | undefined = undefined, depth = 0): any {
+      if (Array.isArray(input)) {
+        return input.map((item: any, index) => {
+          const currentPath = path ? `${path}.${index}` : index.toString();
+          return traverse(item, currentPath, depth + 1);
+        });
+      } else if (input !== undefined && input !== null && typeof input === "object") {
+        const result: any = {};
+
+        for (const key of Object.keys(input)) {
+          const currentPath = path ? `${path}.${key}` : key;
+          result[key] = traverse(input[key], currentPath, depth + 1);
+        }
+
+        if (depth > 1) {
+          const newPath = path?.substring(0, path.length - 2);
```

Comment:
What's the substring for?

Reply:
It's to create a path name that excludes the index, since I just end up building an array of ids.

Reply:
For example: `questions.0.answers.0` -> `questions.0.answers = [1]`

Reply:
I realize the above might be a little confusing haha, in that case the `1` is the id in `questions.0.answers.0`